### DISCUSSION TITLE
SCAM / AIM support in Schematics Pro

### DISCUSSION THREAD

**Discussion by st-clair-clarke** [cyrilletuzi/vscode-angular-schematics#789] [Ideas]
Does schematics pro support Single Component Angular Modules (SCAM) for component, directive, service etc?

**Comment by cyrilletuzi** [comment]
Hi @st-clair-clarke,

Thanks for your interest in Schematics Pro.

For now, it's already possible by installing the SCAM schematics:
```
npm i @ngneat/aim -D
```

Then:
- reload VS Code (the extension does not watch `node_modules` as it would be too expensive)
- right click on the folder of your choice
- "Angular: Generate another schematic"
- choose `@ngneat/aim`
- choose component, directive or pipe

Very soon, it will also be possible by adding your own advanced schematics in a way *much* easier than the current available solutions. I'll also provide a list of ready-to-use recipes, including SCAM / AIM. I'll keep you updated when this alternative is fully implemented.

**Comment by cyrilletuzi** [comment]
As `@ngneat/scam` has been renamed `@ngneat/aim` recently, I just updated Schematics Pro accordingly.

**Comment by st-clair-clarke** [comment]
@ngneat/scam or @ngneat/aim DOES NOT work on my computers - always get the following...

```
Executing task: pnpx ng generate @ngneat/aim:component --name=zom --project=medsoft-preferences-feature --path=libs/medsoft/preferences/feature/src/lib/page/lab-preference-page/lab-preference.page.ts --no-interactive --dry-run <

An unhandled exception occurred: Path "/home/stclair/__/@workspace/pim/node_modules/.pnpm/@ngneat+aim@2.0.0/node_modules/@schematics/angular/component/schema.json" does not exist.
See "/tmp/ng-249Vjl/angular-errors.log" for further details.
The terminal process "bash '-c', 'pnpx ng generate @ngneat/aim:component --name=zom --project=medsoft-preferences-feature --path=libs/medsoft/preferences/feature/src/lib/page/lab-preference-page/lab-preference.page.ts --no-interactive --dry-run'" terminated with exit code: 127.

Terminal will be reused by tasks, press any key to close it.
```

**Comment by cyrilletuzi** [comment]
The issue is not related to this extension. It's due to the fact that you use pnpm (so the dependencies are not always directly under `node_modules`) and that the `@ngneat/aim` schematics collection [references the `@schematics/angular` schemas](https://github.com/ngneat/aim/blob/master/src/collection.json) directly from `node_modules/@schematics/angular`.

That's probably not the only issue you'll have while using pnpm, as it's a nightmare to support for tools authors (so most tools just don't).

For information, **[Schematics Pro](https://www.schematicspro.dev/) bypasses package managers** entirely by having its own internal schematics generator, not relying on `node_modules`. So pnpm is not a problem anymore.

Currently, the internal schematics generator works with the default schematics (classic components, services, modules...). It will soon be working with custom schematics too, so the SCAM / AIM will be possible, even with pnpm.

**Comment by st-clair-clarke** [comment]
Is there a trial period for Schematics PRO. I would lke to try it before I buy.

**Comment by st-clair-clarke** [comment]
Purchased Pro. However, I don't see how to generate SCAM items. I thought you said that @ngneat/aim was incorporated. 
Can't seem to find it.

**Comment by cyrilletuzi** [comment]
Hi @st-clair-clarke,

I sent you an email to help you directly.

**Comment by cyrilletuzi** [comment]
~~For everyone information, proper AIM / SCAM support has been added to [Schematics Pro](https://cyrilletuzi.gumroad.com/l/schematicspro/).~~

With standalone components, SCAM pattern is now useless.
